common/src/main/java/net/caffeinemc/mods/sodium/client/render/chunk/compile/ChunkBuildBuffers.java

```diff
@@ -61,41 +60,45 @@ public BuiltSectionMeshParts createMesh(TerrainRenderPass pass, boolean forceUna
         var builder = this.builders.get(pass);
 
         List<ByteBuffer> vertexBuffers = new ArrayList<>();
-        VertexRange[] vertexRanges = new VertexRange[ModelQuadFacing.COUNT];
+        int[] vertexCounts = new int[ModelQuadFacing.COUNT];
 
-        int vertexCount = 0;
+        int vertexSum = 0;
 
         for (ModelQuadFacing facing : ModelQuadFacing.VALUES) {
+            var ordinal = facing.ordinal();
             var buffer = builder.getVertexBuffer(facing);
 
             if (buffer.isEmpty()) {
+                vertexCounts[ordinal] = -1;
```

Comment:
I think it is unnecessary to use `-1` here, I probably just thought it was necessary since `null` vertex range objects are relevant for my other PR https://github.com/CaffeineMC/sodium-fabric/pull/2421 (which stands in conflict with this one, at least without refactoring it to use some other system of storing the reordering of facings, for example by encoding the positions into a long)

Reply:
I've pushed changes to remove the `-1` and also to improve the handling of the mesh object (instead it now just uses the vertex counts).